dev/tests/api-functional/testsuite/Magento/GraphQl/Newsletter/Customer/SubscribeEmailToNewsletterTest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GraphQl\Newsletter\Guest;
+
+use Magento\Framework\Exception\AuthenticationException;
+use Magento\Integration\Api\CustomerTokenServiceInterface;
+use Magento\Newsletter\Model\ResourceModel\Subscriber as SubscriberResourceModel;
+use Magento\Newsletter\Model\Subscriber;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\TestCase\GraphQlAbstract;
+
+/**
+ * Test newsletter email subscription for customer
+ */
+class SubscribeEmailToNewsletterTest extends GraphQlAbstract
+{
+    /**
+     * @var CustomerTokenServiceInterface
+     */
+    private $customerTokenService;
+
+    /**
+     * @var SubscriberResourceModel
+     */
+    private $subscriberResource;
+
+    protected function setUp()
+    {
+        $objectManager = Bootstrap::getObjectManager();
+        $this->customerTokenService = $objectManager->get(CustomerTokenServiceInterface::class);
+        $this->subscriberResource = $objectManager->get(SubscriberResourceModel::class);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Customer/_files/customer.php
+     */
+    public function testAddCustomerEmailIntoNewsletterSubscription()
+    {
+        $query = $this->getQuery('customer@example.com');
+        $response = $this->graphQlMutation($query, [], '', $this->getHeaderMap());
+
+        self::assertArrayHasKey('subscribeEmailToNewsletter', $response);
+        self::assertNotEmpty($response['subscribeEmailToNewsletter']);
+        self::assertEquals(Subscriber::STATUS_SUBSCRIBED, $response['subscribeEmailToNewsletter']['status']);
+    }
+
+    /**
+     * @return string
+     */
+    private function getQuery(string $email = ''): string
+    {
+        return <<<QUERY
+mutation {
+  subscribeEmailToNewsletter(
+    email: "$email"
+  ) {
+    status
+  }
+}
+QUERY;
+    }
+
+    /**
+     * Retrieve customer authorization headers
+     *
+     * @param string $username
+     * @param string $password
+     * @return array
+     * @throws AuthenticationException
+     */
+    private function getHeaderMap(string $username = 'customer@example.com', string $password = 'password'): array
+    {
+        $customerToken = $this->customerTokenService->createCustomerAccessToken($username, $password);
+        $headerMap = ['Authorization' => 'Bearer ' . $customerToken];
```

Comment:
You don't need the temp `$headerMap` variable here. You may return the array

Reply:
yep!

THank you, @rogyar